File: grmc/representation/entity.py
# Define Annotation Class
import enum
from dataclasses import dataclass

from grmc.representation.util import Identifier


class Emotion(enum.Enum):
    NEUTRAL = 0
    ANGER = 1
    DISGUST = 2
    FEAR = 3
    HAPPINESS = 4
    JOY = 5
    SADNESS = 6
    SURPRISE = 7


class Gender(enum.Enum):
    UNDEFINED = 0
    FEMALE = 1
    MALE = 2
    OTHER = 3


@dataclass
class Instance:
    id: Identifier


@dataclass
class Object(Instance):
    label: str


@dataclass
class Person(Instance):
    name: str
    age: int
    gender: Gender


@dataclass
class Friend(Person):
    pass

File: webapp/src/app/entity.ts
enum Emotion {
  NEUTRAL,
  ANGER,
  DISGUST,
  FEAR,
  HAPPINESS,
  JOY,
  SADNESS,
  SURPRISE,
}

enum Gender {
  UNDEFINED,
  FEMALE,
  MALE,
  OTHER,
}


export interface Instance {
  id: string;
}

export interface Obj extends Instance {
  label: string;
}

export interface Person extends Instance {
  name: string;
  age: number;
  gender: Gender;
}

export interface Friend extends Person {
  // no additional fields
}
